### PR TITLE
build_requirejs refactor and B5 updates

### PR DESCRIPTION
**Pull request by orangejenny**
## Technical Summary
Review by commit. This PR does a few things:
1. Ignores `bootstrap_version` argument in favor of looping over bootstrap versions within the command (first commit). This will allow us to stop [running this command twice during deploy](https://github.com/dimagi/commcare-cloud/blob/25fffbdddcd748fc90ee60a9e1c597706de53612/src/commcare_cloud/ansible/roles/deploy_hq/tasks/staticfiles_collect.yml#L30-L31). Once this is merged, I'll do a [commcare-cloud PR](https://github.com/dimagi/commcare-cloud/pull/6251) to remove the second run, and then [another HQ PR](https://github.com/dimagi/commcare-hq/pull/34349) to drop the `bootstrap_version` param altogether.
2. A lot of readability and organizational improvements (most commits)
3. Drops the `split_bundles` manual config in favor of automatically detecting whether a page is B3 or B5. This is based on B5 pages using a new tag, `requirejs_main_b5`, that's just an alias for `requirejs_main`. All B5 pages should use this tag, until we remove B3 altogether, at which point we can switch back to `requirejs_main`.

### Safety story
I've smoke tested locally and am testing more extensively on staging, checking that:
* Expected build artifacts are created
* B3 pages still load
* B5 pages still load, whether or not they use B5 bootstrap javascript

### Automated test coverage

Not much. The changes to the bootstrap migration tool include tests.

### QA Plan

Targeted developer testing is more appropriate here from QA. Feel free to critique the testing strategy described above.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/hqwebapp/templates/hqwebapp/bootstrap5/base_paginated_crud.html
Comment (orangejenny) [inline corehq/apps/hqwebapp/templates/hqwebapp/bootstrap5/base_paginated_crud.html:7]: Verified that all templates that directly inherit from this one use requirejs, and there are no templates inheriting from those templates.

File: corehq/apps/hqwebapp/management/commands/build_requirejs.py
Comment (MartinRiese) [inline corehq/apps/hqwebapp/management/commands/build_requirejs.py:232]: This would prevent your local git pollution you mentioned in the meeting, right?

File: corehq/apps/hqwebapp/management/commands/build_requirejs.py
Comment (MartinRiese) [inline corehq/apps/hqwebapp/management/commands/build_requirejs.py:109]: one to many `filename =` not sure what that actually does.

**Review by MartinRiese — CHANGES_REQUESTED** [review]
I really appreciate the small commits. They make the review so much easier. I just wish there was a good way to get the moved functions next to each other in the diff so see changes better. If you know a better way than having two windows and scrolling them to the appropriate places, let me know.

File: corehq/apps/hqwebapp/templates/hqwebapp/bootstrap5/base_paginated_crud.html
Comment (biyeun) [reply corehq/apps/hqwebapp/templates/hqwebapp/bootstrap5/base_paginated_crud.html:7]: @orangejenny are those all templates that inherit from the bootstrap3 version? I don't think there are a lot of templates that inherit from the bootstrap5 version right now. just want to double check...

File: corehq/apps/hqwebapp/templatetags/hq_shared_tags.py
Comment (biyeun) [inline corehq/apps/hqwebapp/templatetags/hq_shared_tags.py:641]:
```suggestion
    for that tag, uses this alias to determine which version of Bootstrap a template uses.
```

File: corehq/apps/hqwebapp/management/commands/build_requirejs.py
Comment (biyeun) [inline corehq/apps/hqwebapp/management/commands/build_requirejs.py:22]: I realize this was changed for `bundle.b5` and `bundle.b3` below, but my concern is keeping `bootstrap3` and `bootstrap5` as consistent, searchable strings. Any chance those bundles can be `bundle.bootstrap3` and `bundle.bootstrap3` to keep these strings consistent and discoverable? :)

File: corehq/apps/hqwebapp/templates/hqwebapp/bootstrap5/base_paginated_crud.html
Comment (orangejenny) [reply corehq/apps/hqwebapp/templates/hqwebapp/bootstrap5/base_paginated_crud.html:7]: I didn't even notice that template was split. Yeah, nothing inherits from the B5 version, just the B3. I added b7ebeda8fb0 to remove this code from the B3 version.

File: corehq/apps/hqwebapp/management/commands/build_requirejs.py
Comment (orangejenny) [reply corehq/apps/hqwebapp/management/commands/build_requirejs.py]: sure - 5901b72c441

File: corehq/apps/hqwebapp/management/commands/build_requirejs.py
Comment (orangejenny) [reply corehq/apps/hqwebapp/management/commands/build_requirejs.py:232]: yes, the git pollution only happens when `local` is set

File: corehq/apps/hqwebapp/management/commands/build_requirejs.py
Comment (orangejenny) [reply corehq/apps/hqwebapp/management/commands/build_requirejs.py]: whoops - 7cbf25a7d67

**Comment by orangejenny** [comment]
@MartinRiese I don't even use the side by side diff, I'm used to the single-column one 😲 so no good ideas on how to make review easier.

File: corehq/apps/hqwebapp/management/commands/build_requirejs.py
Comment (orangejenny) [reply corehq/apps/hqwebapp/management/commands/build_requirejs.py:22]: eh, I'm not worried about the discoverability of this particular file. The build pipeline is unique enough that it's hard to think of what general bootstrap task someone would be doing that would affect it.

**Comment by orangejenny** [comment]
(marking don't merge while I investigate some possible bugs)

File: corehq/apps/hqwebapp/management/commands/build_requirejs.py
Comment (millerdev) [inline corehq/apps/hqwebapp/management/commands/build_requirejs.py:279]: Any chance that `basename` could occur more than once in the line? Would it be bad if multiple occurrences were replaced?
```suggestion
                        line += '?version={file_hash}'
```

File: corehq/apps/hqwebapp/management/commands/build_requirejs.py
Comment (millerdev) [inline corehq/apps/hqwebapp/management/commands/build_requirejs.py:275]: What do you think of moving this regex to a constant at the top of the file?
```py
BUNDLE_MAP_PATTERN = re.compile(r'sourceMappingURL=(bundle.b[%s].js.map)$'
                                % "".join(str(v) for v in BOOTSTRAP_VERSIONS))
```
Then this line would change to
```suggestion
                    match = BUNDLE_MAP_PATTERN.search(line)
```

File: corehq/apps/hqwebapp/management/commands/build_requirejs.py
Comment (millerdev) [inline corehq/apps/hqwebapp/management/commands/build_requirejs.py:205]: Is it possible `"build"` could appear more than once in `src`? Would it be a problem if it were replaced more than once?

File: corehq/apps/hqwebapp/management/commands/build_requirejs.py
Comment (millerdev) [inline corehq/apps/hqwebapp/management/commands/build_requirejs.py:141]: This would only ever match `requirejs_main_b5` since `requirejs_main_b3` does not exist, right?

It would make this file easier to update in the future if this and the logic below used values from `BOOTSTRAP_VERSIONS` rather than having them hard-coded.

File: corehq/apps/hqwebapp/management/commands/build_requirejs.py
Comment (millerdev) [reply corehq/apps/hqwebapp/management/commands/build_requirejs.py:46]: Oh, that's a great question. It probably will fail, so that makes my suggestion not great.

File: corehq/apps/hqwebapp/management/commands/build_requirejs.py
Comment (orangejenny) [reply corehq/apps/hqwebapp/management/commands/build_requirejs.py]: 58dafa59cfd - this also combines the requirejs_main regexes into a single constant, which is a nice simplification

File: corehq/apps/hqwebapp/management/commands/build_requirejs.py
Comment (orangejenny) [reply corehq/apps/hqwebapp/management/commands/build_requirejs.py]: yes, I must have written this while planning to introduce both `requirejs_main_b3` and `requirejs_main_b5` and then decided not to do that. So I could hard-code the `5` in the regex. That's less flexible regarding future changes, but realistically, I'm anticipating removing all version-related logic once the bootstrap migration is done.

File: corehq/apps/hqwebapp/management/commands/build_requirejs.py
Comment (orangejenny) [reply corehq/apps/hqwebapp/management/commands/build_requirejs.py]: It's pretty unlikely and would likely make the copy fail, but I tightened it up a bit here: faef74a452d

File: corehq/apps/hqwebapp/management/commands/build_requirejs.py
Comment (orangejenny) [reply corehq/apps/hqwebapp/management/commands/build_requirejs.py]: Followup in 4048d572c89 

Annoyingly and surprisingly, I couldn't get the same regex working in both grep and python, needed to do more escaping for grep. But I also noticed that the old grep expression was only capturing `requirejs_main`, not `requirejs_main_b5`, which was a bug.

Also made the versions in the regex dynamic, which I missed when first reading your comment.

File: corehq/apps/hqwebapp/management/commands/build_requirejs.py
Comment (orangejenny) [reply corehq/apps/hqwebapp/management/commands/build_requirejs.py]: I ultimately updated this to only reference b5, in https://github.com/dimagi/commcare-hq/pull/34346/commits/4048d572c890b63ca22dc84c55eca2c017a7e2da

File: corehq/apps/hqwebapp/management/commands/build_requirejs.py
Comment (orangejenny) [reply corehq/apps/hqwebapp/management/commands/build_requirejs.py]: hmm, it's certainly possible. It'll eventually fail because the path to `r.js` is relative, though that isn't deliberate. I'm inclined to either leave it alone or check the current directory and fail hard if it isn't in the root. Does that seem messy?

**Comment by orangejenny** [comment]
@millerdev Ready for re-review.

@MartinRiese Could you also re-review? GitHub won't be satisfied until request changes switches to approved.

**Comment by orangejenny** [comment]
I'm hoping to merge this tomorrow so it makes Thursday deploy and then hopefully https://github.com/dimagi/commcare-hq/pull/34271 can go out next week (don't want to deploy them together).

File: corehq/apps/hqwebapp/management/commands/build_requirejs.py
Comment (millerdev) [reply corehq/apps/hqwebapp/management/commands/build_requirejs.py]: I prefer tools that have minimal runtime context assumptions. But I understand if that seems like a nice to have and is out of scope for this PR.

**Review by millerdev — APPROVED** [review]

File: corehq/apps/hqwebapp/management/commands/build_requirejs.py
Comment (orangejenny) [reply corehq/apps/hqwebapp/management/commands/build_requirejs.py]: Cool, I will leave it for now and keep that in mind.

**Review by MartinRiese — APPROVED** [review]

**Pull request merged**
